projects/plugins/jetpack/extensions/plugins/ai-assistant-plugin/components/ai-assistant-plugin-sidebar/index.tsx

```diff
@@ -89,6 +94,21 @@ export default function AiAssistantPluginSidebar() {
 	const { checkoutUrl } = useAICheckout();
 
 	const { tracks } = useAnalytics();
+
+	const isViewable = useSelect( select => {
+		const postTypeName = ( select( editorStore ) as typeof EditorSelectors ).getCurrentPostType();
+		// The coreStore select type lacks the getPostType method, so we need to cast it to the correct type
+		const postTypeObject = ( select( coreStore ) as unknown as CoreSelect ).getPostType(
+			postTypeName
+		);
+
+		return postTypeObject?.viewable;
+	}, [] );
+	// If the post type is not viewable, do not render my plugin.
+	if ( ! isViewable ) {
+		return null;
+	}
```

Comment:
We're using `AI Assistant` panel in other places (`Document Settings`, `Pre-Publish`) than `Jetpack Sidebar`, we should trigger an early return and remove it from all places?

Reply:
`Pre-Publish` shouldn't be a concern here since it's not displayed in the site editor. I'm thinking `Document Settings` may still be valuable (and work) in the site editor though; what do you think?

Reply:
@jeherve Thinking better about, don't think any of the current features we have on AI suits good with Site Editor. I think we're good on the early return. If we change the mind in the future we can return and update cc/ @Automattic/jetpack-agora

Reply:
Does this mean we should add early returns elsewhere in addition to what I've added here?

Reply:
@jeherve don't think so, this is the entry for all features that lives on the sidebar.

Reply:
Yeah, current features do not suit the Site Editor. Breve could be an exception later on, but that is for the future us.